### DISCUSSION TITLE
brew bump-formula-pr Error: Query must include 'is:issue' or 'is:pull-request'

### DISCUSSION THREAD

**Discussion by rubenhak** [Homebrew/discussions#4019] [Tap maintenance and Homebrew development]
Hi, I'm trying to publish a version update. Getting the error below. Hard to tell what is going on. Would appreciate some help here. Thank you!

```sh
$ brew bump-formula-pr kubevious --url=https://registry.npmjs.org/kubevious/-/kubevious-1.0.30.tgz  --debug                                                                                                                     

/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/kubevious.rb

/usr/local/Homebrew/Library/Homebrew/shims/shared/git --version

/usr/local/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/3.6.14-8-g4a07924\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 13.0.1\)\ curl/7.84.0 --header Accept-Language:\ en --retry 3 --location [https://api.github.com/search/issues\?q=kubevious\+repo\%3AHomebrew\%2Fhomebrew-core\+state\%3Aopen\+in\%3Atitle\&per_page=100](https://api.github.com/search/issues%5C?q=kubevious%5C+repo%5C%3AHomebrew%5C%2Fhomebrew-core%5C+state%5C%3Aopen%5C+in%5C%3Atitle%5C&per_page=100) --header Accept:\ application/vnd.github\+json --write-out '

'\%\{http_code\} --header Authorization:\ token\ ****** --dump-header /private/tmp/github_api_headers20221205-20769-rpqr7d

Error: Query must include 'is:issue' or 'is:pull-request'

/usr/local/Homebrew/Library/Homebrew/utils/github/api.rb:306:in `raise_error'

/usr/local/Homebrew/Library/Homebrew/utils/github/api.rb:234:in `open_rest'

/usr/local/Homebrew/Library/Homebrew/utils/github.rb:179:in `search'

/usr/local/Homebrew/Library/Homebrew/utils/github.rb:40:in `search_issues'

/usr/local/Homebrew/Library/Homebrew/utils/github.rb:73:in `issues_for_formula'

/usr/local/Homebrew/Library/Homebrew/utils/github.rb:497:in `fetch_pull_requests'

/usr/local/Homebrew/Library/Homebrew/utils/github.rb:506:in `check_for_duplicate_pull_requests'

/usr/local/Homebrew/Library/Homebrew/dev-cmd/bump-formula-pr.rb:413:in `check_open_pull_requests'

/usr/local/Homebrew/Library/Homebrew/dev-cmd/bump-formula-pr.rb:129:in `bump_formula_pr'

/usr/local/Homebrew/Library/Homebrew/brew.rb:95:in `<main>'
```

**Comment by dtrodrigues** [comment]
What's the output of `brew config`, `brew doctor`, and `curl -v https://api.github.com/search/issues\?q=kubevious\+repo\%3AHomebrew\%2Fhomebrew-core\+state\%3Aopen\+in\%3Atitle\&per_page=100` ?

**Comment by eugenesvk** [comment]
According to [GitHub api docs](https://docs.github.com/en/rest/search?apiVersion=2022-11-28#search-issues-and-pull-requests) you need to add an explicit parameter for issues (`is:issue`) and pull requests (`is:pr`) to your url since you can't search for both

> Note: For user-to-server GitHub App requests, you can't retrieve a combination of issues and pull requests in a single query. Requests that don't include the is:issue or is:pull-request qualifier will receive an HTTP 422 Unprocessable Entity response. To get results for both issues and pull requests, you must send separate queries for issues and pull requests. For more information about the is qualifier, see "Searching only issues or pull requests."

Since the top function in your log is `check_open_pull_requests`, I guess you need to search for pull requests only, so you'd need to add `is:pr`, maybe here:
https://github.com/Homebrew/brew/blob/884c4bec14073314ac6e8e9a6a64bc3b26161ab6/Library/Homebrew/utils/github.rb#L489
```ruby
  def fetch_pull_requests(name, tap_remote_repo, state: nil, version: nil)
    if version.present?
      #query= "#{name} #{version}" #OLD
      query = "#{name} #{version} is:pr"
      regex = /(^|\s)#{Regexp.quote(name)}(:|,|\s)(.*\s)?#{Regexp.quote(version)}(:|,|\s|$)/i
    else
      # query = {name} #OLD
      query = "#{name} is:pr"
      regex = /(^|\s)#{Regexp.quote(name)}(:|,|\s|$)/i
    end
```

(this should add the extra `\+is\%3Apr` before `\+repo` in your curl link)

**Comment by anatawa12** [comment]
I'm affecting with this issue so I fixed this issue with this pullreq. https://github.com/Homebrew/brew/pull/14584
